skbio/stats/power.py

```diff
@@ -476,44 +485,34 @@ def subsample_paired_power(test, meta, cat, control_cats, order=None,
     # Checks for the number of sampling pairs avaliable
     sub_ids = paired_subsamples(meta, cat, control_cats, order, strict_match)
 
-    # Determines the minimum number of ids avaliable
-    num_ids = len(sub_ids[0])
-
-    # Checks there are enough samples to subsample
-    if num_ids <= min_observations:
-        raise ValueError('There are not enough samples for subsampling.')
-
-    # Calculates the effect size vector
-    if min_counts is None:
-        min_counts = counts_interval
-
-    if (max_counts - min_counts) < counts_interval:
-        raise ValueError("No subsamples of the specified size can be drawn.")
-
-    sample_counts = np.arange(min_counts,
-                              min(max_counts, num_ids),
-                              counts_interval)
+    ratio, num_p, sample_counts = \
+        _check_subsample_power_inputs(test=test,
+                                      samples=sub_ids,
+                                      draw_mode='matched',
+                                      min_observations=min_observations,
+                                      min_counts=min_counts,
+                                      max_counts=max_counts,
+                                      counts_interval=counts_interval)
 
     # Prealocates the power array
-    power = np.zeros((num_runs, len(sample_counts)))
-
-    power[0, :] = _calculate_power_curve(test,
-                                         sub_ids,
-                                         sample_counts,
-                                         mode="matched",
-                                         num_iter=num_iter,
-                                         alpha=alpha_pwr)
-
-    for id1 in np.arange(1, num_runs):
-        sub_ids = paired_subsamples(meta, cat, control_cats, order,
-                                    strict_match)
-        # Calculates the power curve
-        power[id1, :] = _calculate_power_curve(test,
-                                               sub_ids,
-                                               sample_counts,
-                                               num_iter=num_iter,
-                                               alpha=alpha_pwr,
-                                               mode="matched")
+    power = np.zeros((num_runs, len(sample_counts), num_p))
+
+    # Calculates power instances
```

Comment:
Not with major refactoring of the way `paired_subsamples` returns samples.

Under the current set up, the critical difference is that a paired subsample is drawn for ever iteration. This increases the number of Monte Carlo simulations, and helps to address a bug in the current code.
`paired_subsamples` could be refactored to return all the subsampled pairs, and _calculate_power_curve's draw could be slightly refactored to support the correct drawing, and this could be set up as boiler plate.